File: 3885-count-special-triplets/3885-count-special-triplets.py
class Solution:
       def specialTriplets(self, A: List[int]) -> int:
        n = len(A)
        left, right = Counter(), Counter(A)
        res = 0
        for a in A:
            right[a] -= 1
            res += left[a * 2] * right[a * 2]
            left[a] += 1
        return res % (10 ** 9 + 7)

File: 3885-count-special-triplets/3885-count-special-triplets.java
class Solution {
        public int specialTriplets(int[] A) {
        int mod = 1_000_000_007, res = 0;
        Map<Integer, Integer> left = new HashMap<>(), right = new HashMap<>();
        for (int a : A) {
            right.put(a, right.getOrDefault(a, 0) + 1);
        }
        for (int a : A) {
            right.put(a, right.get(a) - 1);
            int ci = left.getOrDefault(a * 2, 0);
            int ck = right.getOrDefault(a * 2, 0);
            res = (int)((res + 1L * ci * ck) % mod);
            left.put(a, left.getOrDefault(a, 0) + 1);
        }
        return res;
    }
}